internal/pkg/metadb/metadb.go

```diff
@@ -910,7 +910,7 @@ func (m *MetaDB) GetMultiRecords(ctx context.Context, storeKeys, recordKeys []st
 
 func (m *MetaDB) createDatastoreKeys(storeKeys, recordKeys []string) ([]*ds.Key, error) {
 	if len(storeKeys) != len(recordKeys) {
-		return nil, status.Errorf(codes.Internal, "metadb createDatastoreKeys: invalid store/record key array(s)  length")
+		return nil, status.Errorf(codes.InvalidArgument, "metadb createDatastoreKeys: invalid store/record key array(s)  length")
```

Comment:
`status.Error` (no formatting) would be good here? also there's an extra space before length.